packages/gatsby/src/commands/develop.js

```diff
@@ -545,15 +545,15 @@ module.exports = async (program: any) => {
       program.port
     )
     const isSuccessful = !messages.errors.length
-    // if (isSuccessful) {
-    // console.log(chalk.green(`Compiled successfully!`))
-    // }
-    // if (isSuccessful && (isInteractive || isFirstCompile)) {
     if (isSuccessful && isFirstCompile) {
       printInstructions(program.sitePackageJson.name, urls, program.useYarn)
       printDeprecationWarnings()
       if (program.open) {
-        Promise.resolve(openurl(urls.localUrlForBrowser)).catch(err =>
+        const urlToOpen =
+          process.platform === `win32`
```

Comment:
I would fix this in prepareUrls where we convert 0.0.0.0 to localhost. We should also do the isWSL check as we'll have the same issue.

CRA does this too
https://github.com/facebook/create-react-app/blob/eaf0b176e2d2af4f9a1f092caf5fd7356cd73844/packages/react-dev-utils/WebpackDevServerUtils.js#L44